File: 0007-Reverse-Integer/lc0007.js
// JavaScript solution.

/**
 * @param {number} x
 * @return {number}
 */

var reverse = function(x) {
    let ret = 0;
    while (x !== 0) {
        let pop = x % 10;
        x = ~~(x/10);
        if (ret > 214748364 || (ret === 214748364 && pop > 7)) return 0;
        if (ret < -214748364 || (ret === -214748364 && pop < -8)) return 0;
        ret = ret * 10 + pop;
    }
    return ret;
};

File: 0007-Reverse-Integer/lc0007.rs
// Rust solution.

impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut ret: i32 = 0;
        let mut xt: i32 = x;
        while (xt != 0) {
            let pop = xt % 10;
            xt /= 10;
            if (ret > 214748364 || (ret == 214748364 && pop > 7)) { return 0; }
            if (ret < -214748364 || (ret == -214748364 && pop < -8)) { return 0; }
            ret = ret * 10 + pop;
        }
        ret
    }
}
